components/speedreader/rust/lib/src/readability/src/dom.rs

```diff
@@ -190,62 +199,68 @@ pub fn has_nodes(handle: &Handle, tag_names: &[&'static LocalName]) -> bool {
     false
 }
 
+#[inline]
+pub fn has_single_tag_inside_element(handle: &Handle, tag: &LocalName) -> Option<Handle> {
```

Comment:
nit: I'd expect a `has_...()` function to return a boolean. If I understood the function correctly, maybe renaming it to `get_single_element_by_tag()` would be more explicit.